pkg/apis/aws/validation/worker.go

```diff
@@ -89,37 +89,47 @@ func ValidateWorkerConfig(workerConfig *apisaws.WorkerConfig, volume *core.Volum
 func ValidateWorkersAgainstCloudProfileOnCreation(workers []core.Worker, region string, awsCloudProfile *apisaws.CloudProfileConfig, fldPath *field.Path) field.ErrorList {
 	allErrs := field.ErrorList{}
 
-	allErrs = append(allErrs, validateWorkersConfigAgainstCloudProfile(workers, region, awsCloudProfile, fldPath)...)
-
+	for i, w := range workers {
+		allErrs = append(allErrs, validateWorkerConfigAgainstCloudProfile(w, region, awsCloudProfile, fldPath.Index(i))...)
+	}
 	return allErrs
 }
 
-func ValidateWorkersAgainstCloudProfileOnUpdate(_, newWorkers []core.Worker, region string, awsCloudProfile *apisaws.CloudProfileConfig, fldPath *field.Path) field.ErrorList {
+func ValidateWorkersAgainstCloudProfileOnUpdate(oldWorkers, newWorkers []core.Worker, region string, awsCloudProfile *apisaws.CloudProfileConfig, fldPath *field.Path) field.ErrorList {
 	allErrs := field.ErrorList{}
 
 	// Validate the existence of the images the new/updated workers are to use. Validating the images used by old workers is not possible at this point, as they might
 	// have been removed from the CloudProfile already.
-	allErrs = append(allErrs, validateWorkersConfigAgainstCloudProfile(newWorkers, region, awsCloudProfile, fldPath)...)
+	for i, newWorker := range newWorkers {
+		var w core.Worker
+		for _, oldWorker := range oldWorkers {
+			if newWorker.Name == oldWorker.Name {
+				w = oldWorker
+			}
+		}
+		// Validate only new Workers or those whose image has changed.
+		if w.Name == "" || newWorker.Machine.Image != w.Machine.Image {
```

Comment:
You can add a `break` after `w = oldWorker`. 
Potentially you can have a bit more elegant solution by constructing a map[string]core.Worker from the oldWorkers and doing the checks from that:
```
if v, ok := myMap[workerName]; !ok || v.Machine.Image != ...
```

It's an lgtm and the logic seems correct but I had to think couple extra seconds for the `w.Name == ""` in retrospect hence the above proposals.